src/main/kotlin/io/inventi/eventstore/eventhandler/CatchupSubscriptions.kt

```diff
@@ -75,40 +62,49 @@ class CatchupSubscriptions(
                 .let { it - 1 } // -1 because catch-up subscription start event numbers are exclusive
                 .takeUnless { it < 0 } // use null instead of -1 to indicate the start of the stream
 
-        subscriptionCheckpointDao.createIfNotExists(SubscriptionCheckpoint(handler.groupName, handler.streamName, checkpoint))
+        with(handler) {
+            subscriptionCheckpointDao.createIfNotExists(SubscriptionCheckpoint(groupName, streamName, checkpoint))
+            subscriptionInitialPositionDao.createIfNotExists(
```

Comment:
we probably want to call these inserts atomically and use a transaction. You could even use transactionTemplate here.

Reply:
Done